packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx

```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
         }
     }, [error, showError]);
 
-    const handleUpdate = () => {
-        if (hasPassword && password && newPassword) {
-            mutate({
-                password,
-                newPassword,
-            });
-        } else if (!hasPassword && newPassword) {
-            mutate({
-                password: '',
-                newPassword,
-            });
-        } else {
-            showToastError({
-                title: 'Required fields: password and new password',
-                timeout: 3000,
-            });
-        }
-    };
+    const onSubmit = form.onSubmit(({ currentPassword, newPassword }) => {
```

Comment:
I know this is an old code.
im not a big fan of useEffect used like that above...
can you remove useEffect and add `onError` callback to the hook?

Reply:
sure, checking now